File: examples/07_compile_time_validation.ts
function main(): void {

interface Data {
  a: number;
  b: number;
  c: any;
}

  function createData(a: number, b: number, c: any): Data {
    return {
      a: a,
      b: b,
      c: c
    };
  }

  const DATA_SIZE: number = 3;

  const SIZE: number = 24;

  const FIELDS: number = 3;

  const HAS_A: boolean = true;

  const HAS_X: boolean = false;
  console.log(`sizeof=${SIZE}, fields=${FIELDS}`);
  console.log(`has_a=${HAS_A}, has_x=${HAS_X}`);

  const MAX_SIZE: number = 64;

  const SIZE_OK: boolean = true;

  const IS_ALIGNED: boolean = true;
  console.log(`size_ok=${SIZE_OK}, aligned=${IS_ALIGNED}`);

  const MODE: string = "optimized";
  console.log(`mode: ${MODE}`);
}

main();

File: examples/07_compile_time_validation.rs
pub fn main() -> () {
    pub struct Data {
        pub a: i64,
        pub b: i64,
        pub c: [u8; 16],
    }
    pub const SIZE: usize = 24;
    pub const FIELDS: usize = 3;
    pub const HAS_A: bool = true;
    pub const HAS_X: bool = false;
    println!("sizeof={}, fields={}", SIZE, FIELDS);
    println!("has_a={}, has_x={}", HAS_A, HAS_X);
    pub const MAX_SIZE: usize = 64;
    pub const SIZE_OK: bool = true;
    pub const IS_ALIGNED: bool = true;
    println!("size_ok={}, aligned={}", SIZE_OK, IS_ALIGNED);
    pub const MODE: &str = "optimized".to_string();
    println!("mode: {}", MODE);
}
